app/models/user_stats.rb

```diff
@@ -10,7 +10,39 @@ def self.exp_for(user)
     self.stats_for(user).exp
   end
 
+  def activity(date_range = nil)
+    {
+        exercises: {
+            solved_count: organization_exercises
+                              .joins(:assignments)
+                              .where(assignments: { top_submission_status: [:passed, :skipped], submitter: user }.merge_if_present(:submitted_at, date_range))
+                              .count,
+            count: organization_exercises.count},
+
+        messages: {
+            count: messages_in_discussions(date_range).count,
+            approved: messages_in_discussions(date_range).where(approved: true).count}
+    }
+  end
+
   def add_exp!(points)
     self.exp += points
   end
+
+  private
+
+  def messages_in_discussions(date_range = nil)
+    Message.joins(:discussion)
+        .where({sender: user.uid, discussions: { organization: organization }}.merge_if_present(:date, date_range))
+  end
+
+  def organization_exercises
```

Comment:
@flbulgarelli accessing directly through guide usages will be faster here, right?
With @julian-berbel we've been thinking that maybe an organization content caché would improve our performance :monocle_face: 

Again, @faloi, this does not need to be changed, we'll discuss it and change it afterwards!

Reply:
Yup, that is true. We could add an `organization.exercises` that does the same

Reply:
Good news, that method already exists and does what we want :smile: